### ISSUE TITLE
Button styled as link in account header

### ISSUE THREAD

**Issue by frankieroberto** [nhsuk/nhsuk-frontend#1798]
The (relatively new) account section in the header [has the option](https://github.com/nhsuk/nhsuk-frontend/blob/main/packages/nhsuk-frontend/src/nhsuk/components/header/template.njk#L140-L155) to use a <button> element which is styled like a link. This is often used for a "Log out" button - to avoid a potential security issue with using links which might cause users to be logged out without actually clicking the button.

However, when testing using screen readers in the accessibility lab, it confused us that using the shortcode for 'next form element' would jump first to "Log out".

Should we add role="link" to the button to avoid this, on the basis that it looks like a link so should behave like one? If so should it also behave like a link, for example opening a new tab if command-clicked?

Are there any other options?

* Would `<button formaction="/log-out">` behave differently from a button embedded in a form, or not?
* Alternatively we could recommend using a real link which takes you to a confirmation page that has a button on it - but this would have the down side of requiring 2 click to log out

**Comment by frankieroberto** [comment]
Just checked GOVUK One Login, and they have a button styled as a link, with the same `<form><button></form>` elements and no `role` attributes.

<img width="1014" height="242" alt="Image" src="https://github.com/user-attachments/assets/b410b300-4a5a-4c9e-9d38-596c7886dd43" />

**Comment by edwardhorsford** [comment]
Where a button is styled as a link, I wonder if we should be recommending it have `role="link"` applied as an attribute - so the semantics presented to assistive technology match the visual appearance. AFAIK this should let AT like dragon correctly classify it based on how it looks - a use should be able to say 'click link x' and it work.

**Comment by colinrotherham** [comment]
Unlike `role="button"` where we only have to polyfill “space bar to activate”, it might be a lot harder in reverse

E.g. Using assistive software to open in a new tab, copy the link, or add to favourites

I'm not saying we support these features, but assistive software might think (as an idenfied link) that they work

### VoiceOver on Safari

For example, to navigate between links:

* Open Links rotor using Control + Option + Command Left/Right
* Move between all links with Control + Option + Command Up/Down

Once you’ve found one:

* Read out the URL with Control + Option + Shift + U
* Open actions menu for the link with Control + Option + Command + Space

We'd want to test how these work with `role="link"` where an `href` doesn't exists
